### ISSUE TITLE
Support for smart cards/Yubikeys/... via gpg-agent

### ISSUE THREAD

**Issue by lvh** [mozilla/sops#191]
I was hopeful this'd work out of the box because the code suggested it talked to gpg-agent, which is e.g. how SSH works for me. Alas:

```
lvh@plotka ~/t/sops > gpg --encrypt --armor \ 
--recipient 1ABDEC84EC75BB574FD6E955484B605C9491E84D < plaintext > ciphertext.gpg
lvh@plotka ~/t/sops> gpg --decrypt < ciphertext.gpg
### I get a pinentry window here, this uses my Yubikey
gpg: encrypted with 4096-bit RSA key, ID E66A058A, created 2017-01-06
      "Laurens (lvh) Van Houtven <lvh@latacora.com>"
My secret
lvh@plotka ~/t/sops> sops -p 1ABDEC84EC75BB574FD6E955484B605C9491E84D myfile.yaml
Could not decrypt the data key with any of the master keys:
	[GPG]: 1ABDEC84EC75BB574FD6E955484B605C9491E84D:	Reading PGP message failed: openpgp: incorrect key
```

**Comment by autrilla** [comment]
This is strongly related to #189

**Comment by lvh** [comment]
Incidentally, I'm also using gpg2; I'm not sure if these are literally the same issue and really my token has nothing to do with it :)

**Comment by autrilla** [comment]
This has been fixed by #238, as if `x/crypto/openpgp` fails SOPS will just call the GPG binary. It works on my workstation with a yubikey.
